app/robots/CMakeLists.txt

```diff
@@ -53,7 +53,9 @@ if(ICUB_INSTALL_ALL_ROBOTS) # Install all robot-specific files
 else()
 
   if (ROBOT_NAME)
-    add_subdirectory(${ROBOT_NAME})
+    if(IS_DIRECTORY ${ROBOT_DIR})
```

Comment:
I used `ROBOT_DIR` because `IS_DIRECTORY` is well defined only on full directory paths [1], not relative ones as `ROBOT_NAME`. 
[1] : https://cmake.org/cmake/help/v3.0/command/if.html